flink-python/pyflink/datastream/connectors/pulsar.py

```diff
@@ -483,18 +438,17 @@ def set_bounded_stop_cursor(self, stop_cursor: StopCursor) -> 'PulsarSourceBuild
         self._j_pulsar_source_builder.setBoundedStopCursor(stop_cursor._j_stop_cursor)
         return self
 
-    def set_deserialization_schema(self,
-                                   pulsar_deserialization_schema: PulsarDeserializationSchema) \
+    def set_value_only_deserializer(self, deserialization_schema: DeserializationSchema) \
             -> 'PulsarSourceBuilder':
         """
-        DeserializationSchema is required for getting the Schema for deserialize message from
-        pulsar and getting the TypeInformation for message serialization in flink.
+        Sets the :class:`~pyflink.common.serialization.DeserializationSchema` for deserializing the
+        value of Pulsars message.
 
-        We have defined a set of implementations, using PulsarDeserializationSchema#flink_type_info
-        or PulsarDeserializationSchema#flink_schema for creating the desired schema.
+        :param deserialization_schema: the :class:`DeserializationSchema` to use for
+            deserialization.
+        :return: this PulsarSourceBuilder.
         """
-        self._j_pulsar_source_builder.setDeserializationSchema(
-            pulsar_deserialization_schema._j_pulsar_deserialization_schema)
+        self._j_builder.setValueOnlyDeserializer(deserialization_schema._j_deserialization_schema)
```

Comment:
I'm curious why you are trying to call `setValueOnlyDeserializer`. Per my understanding, there is no such interface at all: https://github.com/apache/flink-connector-pulsar/blob/main/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

Reply:
That's just wrong :) Hopefully it's fixed now